test/base/ControllerSpecBase.scala

```diff
@@ -16,9 +16,15 @@
 
 package base
 
-import controllers.actions.{FakeIdentifierAction, IdentifierAction}
+import controllers.actions.{DataRetrievalAction, FakeIdentifierAction, IdentifierAction}
+import repositories.UserAnswersRepository
 
 trait ControllerSpecBase extends SpecBase {
   lazy val authenticatedAction: IdentifierAction =
     FakeIdentifierAction.identifierAction(messagesControllerComponents.parsers.anyContent, "some_external_id")
+
+  lazy val dataRetrievalAction: DataRetrievalAction = injector.instanceOf[DataRetrievalAction]
```

Comment:
`dataRetrievalAction` should make use of `test/controllers/actions/FakeDataRetrievalAction.scala` so you can control the data in UA.